src/t_list.c

```diff
@@ -58,11 +147,22 @@ void *listPopSaver(unsigned char *data, size_t sz) {
 }
 
 robj *listTypePop(robj *subject, int where) {
-    long long vlong;
     robj *value = NULL;
 
-    int ql_where = where == LIST_HEAD ? QUICKLIST_HEAD : QUICKLIST_TAIL;
-    if (subject->encoding == OBJ_ENCODING_QUICKLIST) {
+    if (subject->encoding == OBJ_ENCODING_LISTPACK) {
+        unsigned char *p;
+        unsigned char *vstr;
+        int64_t vlen;
+        unsigned char intbuf[LP_INTBUF_SIZE];
+        int pos = (where == LIST_HEAD) ? 0 : -1;
+
+        p = lpSeek(subject->ptr,pos);
```

Comment:
I was going to change it to`p = lpSeek(subject->ptr,(where == LIST_HEAD) ? 0 : -1);`, and finally found that you are right, it will be more efficient.

Reply:
It is also easier to read. 8-)